### DISCUSSION TITLE
How to intergrate foam?

### DISCUSSION THREAD

**Discussion by Perpetual3134** [InteractiveComputerGraphics/SPlisHSPlasH#244] [Q&A]
Hi everybody!

I'm struggeling to integrate the foam in my renderings (Maya with Arnold)... 
have someone informations how to do this?

My actual Pipeline:

1. Sim the "water" particles
2. Sim the foam
3. Brep for both with splashsurf
4. import as obj sequences in maya
5. water Shader - works fine
6. foam Shader - don't now the right way...

**Comment by janbender** [comment]
We typically import the foam particles directly without surface reconstruction. We then just render them as small white spheres. However, I'm sure that there are also better ways to do this.

Am 8. Mai 2023 10:34:20 MESZ schrieb Perpetual3134 ***@***.***>:

**Reply by Perpetual3134** [reply]
Thanks for your help!
One more question:
actually the foamgenerator gives me an output if there are any foam particles.
So there will be a gap at the beginning and the foam particles are not going synchronous to the fluid particles.

Is there a way to force the foamgenerator to output also files with no particles?

**Comment by janbender** [comment]
No, this is not implemented since I don't see a good reason to generate empty files. Since all files have a frame number, it should be easy to sync them.
